### ISSUE TITLE
Early Stop

### ISSUE THREAD

**Issue by stanley-tsgs** [Nixtla/neuralforecast#1108]
### Description

 I got below error when follow Marco's instruction: All models have the parameter early_stop_patience_steps. By default it's -1, meaning it will not early stop. If you set it to 3, then training stop if the validation loss doesn't improve three times in a row.

ERROR - An error occurred: Early stopping conditioned on metric ptl/val_loss which is not available. Pass in or modify your EarlyStopping callback to use any of the following: train_loss, train_loss_step, train_loss_epoch.

**Comment by marcopeix** [comment]
This is because, by default, early stopping requires a validation set. You can create a validation set by specifying the `val_size` parameter in the `fit` method, or by specifying both `val_size` and `test_size` when using `cross_validation`. 

Same as in [this issue](https://github.com/Nixtla/neuralforecast/issues/688).

**Comment by stanley-tsgs** [comment]
@marcopeix Hi Marco, thx for the reply. It turns out I need to setup "val_check_steps" as well along with the "early_stop_patience_steps".
